src/yunohost/service.py

```diff
@@ -708,15 +731,47 @@ def _tail(file, n):
                 pos = f.tell()
                 lines = f.read().splitlines()
 
-                if len(lines) >= to_read or pos == 0:
+                if len(lines) >= to_read:
                     return lines[-to_read:]
 
                 avg_line_length *= 1.3
+        f.close()
 
     except IOError as e:
         logger.warning("Error while tailing file '%s': %s", file, e, exc_info=1)
         return []
 
+    if len(lines) < to_read:
+        previous_log_file = _find_previous_log_file(file)
+        if previous_log_file is not None:
+            lines = _tail(previous_log_file, to_read - len(lines)) + lines
+    return lines[-to_read:offset and -offset or None]
```

Comment:
Got a critical error here because offset was undefined. Not sure what the proper fix is, but in the meantime i just replaced the line with `return lines` instead.

Reply:
That's a conflict with master code because I've removed the notion of offset mostly everywhere because we never used it.